### ISSUE TITLE
BE: Topics: Validate ISR/replication upon creation

### ISSUE THREAD

**Issue by Haarolean** [kafbat/kafka-ui#103]
It's currently possible to edit a topic so the min iSR > replication factor.

**Comment by abhishekray323** [comment]
Hi @Haarolean , 
can you please assign me this issue?

I just want to confirm what this issue is: min.insync.replicas should not be allowed to exceed replication factor as it doesn't make any sense. So there should be validation check that prohibits this while creating and updating the topic, right ?

**Comment by Haarolean** [comment]
> Hi [@Haarolean](https://github.com/Haarolean) , can you please assign me this issue?
> 
> I just want to confirm what this issue is: min.insync.replicas should not be allowed to exceed replication factor as it doesn't make any sense. So there should be validation check that prohibits this while creating and updating the topic, right ?

I remember there could be some other related things, so you might find some other inconsistencies, so please feel free to take a look!

**Comment by Haarolean** [comment]
@abhishekray323 any luck?

**Comment by abhishekray323** [comment]
yeah I was going through all the configurations one by one, it took me some time. Here are the constraints I'm able to come up, will code them now. Let me know, if you have any suggestions on following constraints :

Inconsistencies found:

1. Min.insync.replicas < replication factor
2. "no of out of sync replica" is not set to zero if "unclean.leader.election.enable" is set to true, because enabling "unclean.leader.election.enable" kind of gives idea to developer that I have some out of sync replicas in case any mishappenings happen
3.  Compression.zstd.level: it should be set only when compression.type is set to “zstd”, but we are setting it’s value even when compression value is not set to “zstd”
Similarly, compression.gzip.level, and compression.lz4.level should be set when compression.type is set to them.

4.  if cleanup.policy is set to "compact" only then , the compaction configuration values like min.cleanable.dirty.ratio, min.compaction.lag.ms and max.compaction.lag.ms matters 
5. [delete.retention.ms] should be set only if cleanup.policy=”compact”

6. [segment.ms]≤ local.retention.ms ≤ [retention.ms] : It helps prevent confusing configurations that might not behave as the user expects.
7. Retention.bytes >=Local.retention.bytes >= segment.bytes >= max.message.bytes: 
the segment size should be greater than the individual message size. Although in UI not able to change “ max.message.bytes” value

- If retention enforcement sees that local.retention.bytes is smaller than a single segment, and there’s no room for even one, it will immediately: Mark it for deletion, OR Tier it to remote storage, if enabled. You will have zero local segments, despite having some bytes configured. Kafka can’t do partial segments.
-retention.bytes governs the total storage (local + remote), so it must be ≥ local.retention.bytes
